File: Entities/ProductDetail.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace InteriorDesign.Entities
{
    public class ProductDetail
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        public int ProductId { get; set; }
        public string Detail { get; set; }
        public bool IsDeleted { get; set; }
        public DateTime? DeletedAt { get; set; }
    }
}

File: ClientApp/src/models/productDetail.ts
export class ProductDetailModel {
    id: number;
    productId: number;
    detail: string;
    isDeleted: boolean;
    deletedAt: Date | null;

    constructor() {
        this.id = 0;
        this.productId = 0;
        this.detail = '';
        this.isDeleted = false;
        this.deletedAt = null;
    }
}
